tls/tls.go

```diff
@@ -0,0 +1,284 @@
+package tls
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+
+	"filippo.io/edwards25519"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+// certificate validity in days
+const CERTIFICATE_VALIDITY = 365
+
+type (
+	Key struct {
```

Comment:
// Key is the struct for an edwards representation point

Reply:
added comment